src/Microsoft.DotNet.Build.Tasks.Packaging/test/CreateTrimDependencyGroupsTests.cs

```diff
@@ -311,7 +312,39 @@ public void AddInboxFrameworkGroupsAndDependencies()
             // The only added dependency in portable45-net45+win8+wpa81 is System.Collections.Immutable
             Assert.Equal("System.Collections.Immutable", task.TrimmedDependencies.Where(f => f.GetMetadata("TargetFramework").Equals("portable45-net45+win8+wpa81")).Single().ItemSpec);
         }
+        
+        [Fact]
+        public void AddInboxEmptyFrameworkGroups()
+        {
+            ITaskItem[] files = new[]
+            {
+                CreateFileItem(@"E:\foo\bar.dll", "lib/netstandard1.1", "netstandard1.1")
+            };
+            ITaskItem[] dependencies = new[]
+            {
+                CreateDependencyItem(@"System.Runtime", "4.0.0", "netstandard1.1"),
+                CreateDependencyItem(@"System.Resources.ResourceManager", "4.0.0", "netstandard1.1"),
+                CreateDependencyItem(@"System.IO", "4.0.0", "netstandard1.1"),
+                CreateDependencyItem(@"System.Collections", "4.0.0", "netstandard1.1")
+            };
 
+            CreateTrimDependencyGroups task = new CreateTrimDependencyGroups()
+            {
+                BuildEngine = _engine,
+                Files = files,
+                Dependencies = dependencies,
+                PackageIndexes = packageIndexes
+            };
+
+            _log.Reset();
+            task.Execute();
+            Assert.Equal(0, _log.ErrorsLogged);
+            Assert.Equal(0, _log.WarningsLogged);
+            Assert.Equal(1, task.TrimmedDependencies.Length);
```

Comment:
NIT: do you have to check the length? Wouldn't Single() throw an exception if that is not the case?

Reply:
Sure, I just prefer to fail with an assert.